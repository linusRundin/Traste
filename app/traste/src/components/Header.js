import React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';

import logo from '../assets/traste_logo.png';
import name from '../assets/traste_name.png';
import KeyboardBackspaceIcon from '@mui/icons-material/KeyboardBackspace';

import {Colors} from '../assets/Colors';

/**
 * The main header for the app. Used to navigate the app.
 * @param {goBackHandler} props Contains the handler for navigating back
 * in the document.
 * @return {AppBar} with logo and title.
 */
function Header({goBackHandler}) {
  return (
    <AppBar
      data-testid='header'
      position='sticky'
      sx={{backgroundColor: Colors.trasteNavyBlue}}
    >
      <Toolbar sx={{justifyContent: 'space-between'}}>
        <KeyboardBackspaceIcon fontSize='large' onClick={goBackHandler} />
        <img src={name} alt='traste name' style={{height: '25px'}} />
        <img src={logo} alt='traste logo' style={{height: '40px'}} />
      </Toolbar>
    </AppBar>
  );
}

Header.propTypes = {
  goBackHandler: PropTypes.func.isRequired,
};

export default Header;
