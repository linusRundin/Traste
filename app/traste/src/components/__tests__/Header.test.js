import {render, screen, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '../Header';
import renderer from 'react-test-renderer';
import React from 'react';

afterEach(() => {
  cleanup();
});

/**
   * Test function.
   */
function goBack() {
  console.log('Test back');
}

test('Should not contain SIMON text', () => {
  render(<Header goBackHandler={goBack} />);

  const headerElement = screen.getByTestId('header');
  expect(headerElement).toBeInTheDocument();
  expect(headerElement).not.toHaveTextContent('SIMON');
});

test('Matches snapshot', () => {
  const tree = renderer.create(<Header goBackHandler={goBack} />).toJSON();
  expect(tree).toMatchSnapshot();
});
