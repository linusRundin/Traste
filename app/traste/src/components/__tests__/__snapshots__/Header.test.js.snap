// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-1n32zbc-MuiPaper-root-MuiAppBar-root"
  data-testid="header"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1im6ja8-MuiToolbar-root"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
      data-testid="KeyboardBackspaceIcon"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
      />
    </svg>
    <img
      alt="traste name"
      src="traste_name.png"
      style={
        Object {
          "height": "25px",
        }
      }
    />
    <img
      alt="traste logo"
      src="traste_logo.png"
      style={
        Object {
          "height": "40px",
        }
      }
    />
  </div>
</header>
`;
